src/compiler/checker.ts

```diff
@@ -1642,6 +1642,11 @@ namespace ts {
                 return;
             }
 
+            if (moduleReference.substr(0, 7) === "@types/") {
+                const diag = Diagnostics.Cannot_import_type_declaration_files_Consider_importing_0_instead_of_1;
+                error(errorNode, diag, moduleReference.substr(7), moduleReference);

```

Comment:
Add a `removePrefix` helper like in #15545. (At the same place and with the same text, so that these PRs can merge together.)